pkg/resources/user_password_policy_attachment.go

```diff
@@ -0,0 +1,169 @@
+	passwordPolicy := sdk.NewSchemaObjectIdentifier(passwordPolicyDatabase.Name(), passwordPolicySchema.Name(), passwordPolicyName.Name())
+
+	err := client.Users.Alter(ctx, userName, &sdk.AlterUserOptions{
+		Set: &sdk.UserSet{
+			PasswordPolicy: passwordPolicy,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_database", passwordPolicyDatabase.Name()); err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_schema", passwordPolicySchema.Name()); err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_name", passwordPolicyName.Name()); err != nil {
+		return err
+	}
+	if err := d.Set("user_name", helpers.EncodeSnowflakeID(userName)); err != nil {
+		return err
+	}
+	d.SetId(fmt.Sprintf(`%s|%s`, helpers.EncodeSnowflakeID(passwordPolicy), helpers.EncodeSnowflakeID(userName)))
+
+	return ReadUserPasswordPolicyAttachment(d, meta)
+}
+
+func ReadUserPasswordPolicyAttachment(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
+	userName := sdk.NewAccountObjectIdentifierFromFullyQualifiedName(d.Get("user_name").(string))
+	policyReferences, err := client.PolicyReferences.GetForEntity(ctx, &sdk.GetForEntityPolicyReferenceRequest{
+		// Note: I cannot insert both single and double quotes in the SDK, so for now I need to do this
+		RefEntityName:   sdk.String(userName.FullyQualifiedName()),
+		RefEntityDomain: sdk.String("user"),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Note: this should never happen, but just in case: so far, Snowflake only allows one Password Policy per user.
+	if len(policyReferences) > 1 {
+		return fmt.Errorf("internal error: multiple policy references attached to a user. This should never happen")
+	}
+
+	// Note: this means the resource has been deleted outside of Terraform.
+	if len(policyReferences) == 0 {
+		d.SetId("")
+		return nil
+	}
+	if err := d.Set("password_policy_database", sdk.NewAccountIdentifierFromFullyQualifiedName(policyReferences[0].PolicyDb).Name()); err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_schema", sdk.NewAccountIdentifierFromFullyQualifiedName(policyReferences[0].PolicySchema).Name()); err != nil {
+		return err
+	}
+	if err := d.Set("password_policy_name", sdk.NewAccountIdentifierFromFullyQualifiedName(policyReferences[0].PolicyName).Name()); err != nil {
+		return err
+	}
+	return err
+}
+
+// DeleteAccountPasswordPolicyAttachment implements schema.DeleteFunc.
+func DeleteUserPasswordPolicyAttachment(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
+
+	userName := sdk.NewAccountObjectIdentifierFromFullyQualifiedName(d.Get("user_name").(string))
+
+	err := client.Users.Alter(ctx, userName, &sdk.AlterUserOptions{
+		Unset: &sdk.UserUnset{
+			PasswordPolicy: sdk.Bool(true),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
```

Comment:
after you drop the resource, you have to set the id to the empty string `d.SetId("")`, so the Terraform will know it was successfully removed.